pkg/storage/snapshot/restore.go

```diff
@@ -1007,6 +1141,26 @@ func (ctrl *VMRestoreController) deleteObsoleteDataVolumes(vmRestore *snapshotv1
 		}
 	}
 
+	// Garbage-collect old backend PVC if needed.
+	// Target should always exist at this point, just nil check for safety.
+	if target.Exists() && backendstorage.IsBackendStorageNeededForVM(target.VirtualMachine()) {
```

Comment:
I think it would be better. to have this logic in a separate function

Reply:
Okay added this in a separate function, ack